src/sensors/d200.hpp

```diff
@@ -0,0 +1,157 @@
+  0xf5, 0x1f, 0x52, 0x85, 0xc8, 0x66, 0x2b, 0xfc, 0xb1, 0xed, 0xa0, 0x77,
+  0x3a, 0x94, 0xd9, 0x0e, 0x43, 0xb6, 0xfb, 0x2c, 0x61, 0xcf, 0x82, 0x55,
+  0x18, 0x44, 0x09, 0xde, 0x93, 0x3d, 0x70, 0xa7, 0xea, 0x3e, 0x73, 0xa4,
+  0xe9, 0x47, 0x0a, 0xdd, 0x90, 0xcc, 0x81, 0x56, 0x1b, 0xb5, 0xf8, 0x2f,
+  0x62, 0x97, 0xda, 0x0d, 0x40, 0xee, 0xa3, 0x74, 0x39, 0x65, 0x28, 0xff,
+  0xb2, 0x1c, 0x51, 0x86, 0xcb, 0x21, 0x6c, 0xbb, 0xf6, 0x58, 0x15, 0xc2,
+  0x8f, 0xd3, 0x9e, 0x49, 0x04, 0xaa, 0xe7, 0x30, 0x7d, 0x88, 0xc5, 0x12,
+  0x5f, 0xf1, 0xbc, 0x6b, 0x26, 0x7a, 0x37, 0xe0, 0xad, 0x03, 0x4e, 0x99,
+  0xd4, 0x7c, 0x31, 0xe6, 0xab, 0x05, 0x48, 0x9f, 0xd2, 0x8e, 0xc3, 0x14,
+  0x59, 0xf7, 0xba, 0x6d, 0x20, 0xd5, 0x98, 0x4f, 0x02, 0xac, 0xe1, 0x36,
+  0x7b, 0x27, 0x6a, 0xbd, 0xf0, 0x5e, 0x13, 0xc4, 0x89, 0x63, 0x2e, 0xf9,
+  0xb4, 0x1a, 0x57, 0x80, 0xcd, 0x91, 0xdc, 0x0b, 0x46, 0xe8, 0xa5, 0x72,
+  0x3f, 0xca, 0x87, 0x50, 0x1d, 0xb3, 0xfe, 0x29, 0x64, 0x38, 0x75, 0xa2,
+  0xef, 0x41, 0x0c, 0xdb, 0x96, 0x42, 0x0f, 0xd8, 0x95, 0x3b, 0x76, 0xa1,
+  0xec, 0xb0, 0xfd, 0x2a, 0x67, 0xc9, 0x84, 0x53, 0x1e, 0xeb, 0xa6, 0x71,
+  0x3c, 0x92, 0xdf, 0x08, 0x45, 0x19, 0x54, 0x83, 0xce, 0x60, 0x2d, 0xfa,
+  0xb7, 0x5d, 0x10, 0xc7, 0x8a, 0x24, 0x69, 0xbe, 0xf3, 0xaf, 0xe2, 0x35,
+  0x78, 0xd6, 0x9b, 0x4c, 0x01, 0xf4, 0xb9, 0x6e, 0x23, 0x8d, 0xc0, 0x17,
+  0x5a, 0x06, 0x4b, 0x9c, 0xd1, 0x7f, 0x32, 0xe5, 0xa8
+};
+
+/// @brief serial port used by LiDAR
+#define LIDAR_SERIAL Serial1
+
+/// @brief points per D200 data packet
+const int D200_POINTS_PER_PACKET = 12;
+
+/// @brief number of packets stored teensy-side
+const int D200_NUM_PACKETS_CACHED = 8;
+
+/// @brief baud rate of LiDAR module
+const int D200_BAUD = 230400;
+
+/// @brief start character for packet
+const int D200_START_CHAR = 0x54;
+
+/// @brief frame character for data packet
+const int D200_FRAME_CHAR = 0x2c;
+
+/// @brief length of data packet (bytes)
+const int D200_DATA_PACKET_LEN = 47;
+
+/// @brief length of command packet (bytes)
+const int D200_CMD_PACKET_LEN = 8;
+
+/// @brief hard-coded start command
+const uint8_t D200_START_CMD[] = { 0x54, 0xa0, 0x04, 0, 0, 0, 0, 0x5e };
+
+/// @brief hard-coded stop command
+const uint8_t D200_STOP_CMD[] = { 0x54, 0xa1, 0x04, 0, 0, 0, 0, 0x4a };
+
+/// @brief default scanning speed (rad/s)
+const float D200_DEFAULT_SPEED = (float)(6 * 360) * M_PI / 180.0;
+
+/// @brief min specified scanning speed (rad/s)
+const float D200_MIN_SPEED = (float)(2 * 360 + 1) * M_PI / 180.0;
+
+/// @brief max specified scanning speed (rad/s)
+const float D200_MAX_SPEED = (float)(8 * 360 - 1) * M_PI / 180.0;
+
+struct LidarDataPacket {
+  /// @brief speed of lidar module (rad/s)
+  float lidar_speed = 0.0;
+  
+  /// @brief start angle of measurements (rad)
+  float start_angle = 0.0;
+  
+  /// @brief array of point measurements
+  struct {
+    /// @brief distance (m)
+    float distance = 0.0;
+    /// @brief intensity of measurement. units are ambiguous (not documented)
```

Comment:
I read the docs. It says: 

"the higher the intensity, the larger the signal strength value; the lower the intensity, the smaller the signal strength value." Can this be added?

```suggestion
    /// @brief intensity of measurement. The higher the intensity, the larger the signal strength value.
```

Reply:
It's likely it's just bound by the 1-byte limit unless you found otherwise via testing.